detox/android/detox/src/full/java/com/wix/detox/espresso/matcher/IsDisplayingAtLeastDetox.kt

```diff
@@ -0,0 +1,144 @@
+            .appendText(" and view.getGlobalVisibleRect() covers at least ")
+            .appendValue(areaPercentage)
+            .appendText(" percent of the view's area)")
+    }
+
+    override fun matchesSafely(item: View, mismatchDescription: Description): Boolean {
+        // First check if the view is visible using the default matcher
+        if (!ViewMatchers.isDisplayingAtLeast(areaPercentage).matches(item)) {
+            return false
+        }
+
+        // In case it is, we need to check if the view is actually visible by running intersection of all parent views rects
+        return isDisplayingAtLeast(item, mismatchDescription)
+    }
+
+    private fun isDisplayingAtLeast(view: View, mismatchDescription: Description): Boolean {
+        val viewVisibleRect = getGlobalVisibleRectWorkaround(view)
+        val maxArea = getViewMaxArea(view)
+        val visibleArea: Double = getViewVisibleArea(viewVisibleRect)
+        val displayedPercentage = (visibleArea / maxArea * 100).toInt()
+
+        if (displayedPercentage < areaPercentage) {
+            mismatchDescription
+                .appendText("view was ")
+                .appendValue(displayedPercentage)
+                .appendText(" percent visible to the user")
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Calculate the actual visible area of the view.
+     */
+    private fun getViewVisibleArea(viewVisibleRect: Rect): Double {
+        return (viewVisibleRect.height() * viewVisibleRect.width()).toDouble()
+    }
+
+    /**
+     * Calculate the maximum possible area of the view (including the not visible part).
+     */
+    private fun getViewMaxArea(view: View): Double {
+        val (viewWidth, viewHeight) = getViewSize(view)
+        return viewWidth * viewHeight
+    }
+
+    /**
+     * Calculate the actual size of the view, taking into account the scaling factor and the screen size.
+     */
+    private fun getViewSize(view: View): Pair<Double, Double> {
+        val screenSize = getScreenWithoutStatusBarActionBar(view)
+        val viewWidth = min((view.width * abs(view.scaleX.toDouble())), screenSize.width().toDouble())
+        val viewHeight = min((view.height * abs(view.scaleY.toDouble())), screenSize.height().toDouble())
+        return Pair(viewWidth, viewHeight)
+    }
+
+    /**
+     * Traverse the view hierarchy and calculate the intersection of all parent views with the given view.
+     *
+     * @return The actual visible rectangle of the view.
+     */
+    private fun getGlobalVisibleRectWorkaround(view: View): Rect {
+        var parent = view.parent as? View
+
+        val viewVisibleRect = Rect()
+        view.getGlobalVisibleRect(viewVisibleRect)
+
+        while (parent != null) {
+            val parentVisibleRectangle = Rect()
+            // Fill the visible rectangle of the parent view
+            parent.getGlobalVisibleRect(parentVisibleRectangle)
```

Comment:
same, use also

Reply:
That's an anti-pattern that they call it `get...` but use it as an in-place change method (should've been `fill...`) 😅 🤷🏼‍♂️